src/components/Datatable/Datatable.tsx

```diff
@@ -86,14 +86,29 @@ const Datatable = <D extends Record<string, unknown>>({
   const [hasAppliedFilters, setHasAppliedFilters] = useState<boolean>(
     any(propEq('isApplied', true), filtersState),
   );
+  const [appliedFilters, setAppliedFilters] = useState(filtersState);
 
   useEffect(() => {
     setPageCount(Math.ceil(totalDataSize / defaultPageSize));
   }, [totalDataSize, defaultPageSize]);
 
+  const handleOnDataFetch = useCallback(
+    (pageIndex, pageSize, sortBy, query) => {
```

Comment:
maybe remove the `query` argument, it won't be passed from the Table either cause the same reasons